scully/utils/defaultAction.ts

```diff
@@ -62,3 +65,16 @@ async function doChunks(dataRoutes) {
     return x.concat(activeChunk);
   }, Promise.resolve([]));
 }
+
+function addExtraRoutes(appRoutes, config){
```

Comment:
instead of simply allowing people to provide a `string[]` as the type of `config.extraRotues`, we could allow the type of extra routes to be (string | string[] | Promise<string> | Promise<string[]>)[], which we could just flatten it all and then turn them all into promises, and run a `Promises.all([flattenedAndPromisifiedExtraRoutes])` and then return them.

That is likely the most extensible and support the most amount of people. But... for now, I support this list of strings gets us quite a ways.

Reply:
I would prefer to have the promise<string[]>, as that is indeed the most extensible.
For now, I can live with this. I probably need to do a bit of redesign anyway, so we can start generating "pages" before we are done collecting all routes anyway.